### ISSUE TITLE
feat: add role attribute to button

### ISSUE THREAD

**Issue by LuisRamirezLucio** [microsoft/fast#4878]
# 🙋 Feature Request

Add the option to assign a Role to the buttons.

## 🤔 Expected Behavior
Role can be assigned on buttons

## 😯 Current Behavior
There's no Role on button.

## 💁 Possible Solution
Add a parameter for Role and assign it to the button if it have any value.

## 🔦 Context

Adding the role to the button would add the option to set it as if it where an item in a list and that would affect the behavior of the narrator for non-sighted users.

## 💻 Examples

For the list of feedback options if the button have a role of menuItem narrator would give feedback about the amount of options
![image](https://user-images.githubusercontent.com/85639129/122574433-7f5a4100-d004-11eb-9118-73676efbc358.png)

**Comment by LuisRamirezLucio** [comment]
This has been open for a while, is there any update?

**Comment by chrisdholt** [comment]
I think in a case like this there are several other, likely better options than overriding the role of the component within the design system. Closing this with the following thoughts and recommendation:

- The illustration above where the desired behavior is a menu is likely better accomplished by composing a new instance of Menu Items with a visual style that matches. 
- Allowing an alternative role sets the table for needing to add alternative functionality. Using the example again of a menu item, there are fundamentally different attributes necessary to support in this case as well and we'd need to delegate those to the internal button. Because those don't exist in the template, we'd need to add a way to support arbitrary attributes. That seems more problematic and convoluted to me than simply styling the menu item to look the way you need for a one off.
- If we add this for button we likely need to do so for every other instance of component, which kind of eliminates the point of the `fast-foundation` components while requiring a lot more complexity from each of them.

Ultimately, I think the underlying ask here is to make something which *is* a button and looks like a button behave like a menu. In reality, the way we have structured things, in an effort to provide OOTB accessible components and patterns, is to provide a baseline for an accessible and functional control which can be easily styled. 

TLDR; I think this is supported, but our recommendation is likely the inverse of your ask - style what it is rather than change what it is because of what it looks like.
